rust_src/src/buffers.rs

```diff
@@ -871,6 +871,57 @@ pub fn force_mode_line_update(all: LispObject) -> LispObject {
     all
 }
 
+/// Return a list of overlays which is a copy of the overlay list
+/// LIST, but for buffer BUFFER.
+#[no_mangle]
+pub extern "C" fn copy_overlays(
+    buffer: *mut Lisp_Buffer,
+    list: *mut Lisp_Overlay,
+) -> *mut Lisp_Overlay {
+    if list == ptr::null_mut() {
+        return list;
+    }
+
+    let overlays = LispOverlayRef::from_ptr(list as *mut c_void)
+        .unwrap_or_else(|| panic!("Invalid overlay reference."));
```

Comment:
I'm not sure whether this is important, but the C code checks the markers of each overlay. Maybe we should too ?

Reply:
We could put the assertions in the overlay iterator.